src/sd/spi.rs

```diff
@@ -0,0 +1,174 @@
+use crate::sys::*;
+
+use core::ops::Deref;
+
+use esp_idf_hal::{gpio, peripheral, spi};
+
+pub struct SpiDeviceBuilder(sdspi_device_config_t);
+
+impl From<sdspi_device_config_t> for SpiDeviceBuilder {
+    fn from(config: sdspi_device_config_t) -> Self {
+        Self(config)
+    }
+}
+
+impl Default for SpiDeviceBuilder {
+    fn default() -> Self {
+        Self(sdspi_device_config_t {
+            host_id: spi_host_device_t_SPI2_HOST,
+            gpio_cs: gpio_num_t_GPIO_NUM_13,
+            gpio_cd: gpio_num_t_GPIO_NUM_NC,
+            gpio_wp: gpio_num_t_GPIO_NUM_NC,
+            gpio_int: gpio_num_t_GPIO_NUM_NC,
+        })
+    }
+}
+
+impl SpiDeviceBuilder {
+    pub fn set_spi<T: spi::Spi>(&mut self, _: &mut T) -> &mut Self {
```

Comment:
i think we can make this more explicit to be usable with SPI2 ( and SPI3 where supported) by replacing spi::SPI with the [SpiAnyPins](https://github.com/esp-rs/esp-idf-hal/blob/master/src/spi.rs#L73) trait.

A second addition would be to also take spi peripheral as an argument from our Peripherals struct e.g ` _spi: impl Peripheral<P = SPI> + 'd` lile we do in the spi driver, so a user cannot instantiate it two times.

Reply:
Yes, that makes sense. I've attempted to fix the issue, but I'm not sure if it's the best solution. I will check tomorrow for gated compilation. Thank your advice.